File: Mar-2026/09-PalindromeNumber.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        sx = str(x)
        for i in range(len(sx)//2):
            if (sx[i] != sx[len(sx) - 1 - i]):
                return False
        return True
    
if __name__ == "__main__":
    demo = Solution()
    print(demo.isPalindrome(12321))
    print(demo.isPalindrome(-121))
    print(demo.isPalindrome(12))
    print(demo.isPalindrome(0))

File: Mar-2026/09-PalindromeNumber.cc
#include <iostream>

using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        string sx = to_string(x);
        for (size_t i = 0; i < sx.size(); i++) {
            if (sx[i] != sx[sx.size() - 1 - i]) {
                return false;
            }
        }
        return true;
    }
};

int main() {
    Solution demo = Solution();
    printf("%b\n", demo.isPalindrome(12321));
    printf("%b\n", demo.isPalindrome(-121));
    printf("%b\n", demo.isPalindrome(0));
}